redisinsight/api/src/modules/database-recommendation/scanner/strategies/rts.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { AbstractRecommendationStrategy }
+  from 'src/modules/database-recommendation/scanner/strategies/abstract.recommendation.strategy';
+import { ZSetMemberDto } from 'src/modules/browser/dto';
+import { getUTF8FromBuffer } from 'src/utils/cli-helper';
+import { checkTimestamp } from '../utils';
+
+const maxPercentage = 95;
+
+export class RTSStrategy extends AbstractRecommendationStrategy {
+  /**
+   * Check rts recommendation
+   * @param members
+   */
+
+  async isRecommendationReached(
+    members: ZSetMemberDto[],
+  ): Promise<boolean> {
+    const timestampMemberNames = members.filter(({ name }) => checkTimestamp(getUTF8FromBuffer(name as Buffer)));
+    if ((timestampMemberNames.length / members.length) * 100 > maxPercentage) {
+      return true;
+    }
+    const timestampMemberScores = members.filter(({ score }) => checkTimestamp(String(score)));
+    if ((timestampMemberScores.length / members.length) * 100 > maxPercentage) {
+      return true;
+    }
+    return false;
+  }
+}
```

Comment:
Well you don't have try {} catch block here when here https://github.com/RedisInsight/RedisInsight/pull/1864/files#diff-bdb486bf8c337af570bb02737c4de4cab46efb99889cde98fb4801a9c0b9287fR44 you have.
I thought that you have try catch on another level, where isRecommendationReached invoked. If you don't - please add it on the top level (most ptobably on provider or service level) to not care about it in all strategies.

Reply:
I will put it in scanner level
```
try {
      const isRecommendationReached = await strategy.isRecommendationReached(data);

      if (isRecommendationReached) {
        return { name };
      }
    } catch (err) {
      // ignore errors
      return null;
    }
   ```